src/crypto/sha512.cpp

```diff
@@ -92,7 +92,8 @@ namespace fc {
    void from_variant( const variant& v, sha512& bi, uint32_t max_depth )
    {
       std::vector<char> ve = v.as< std::vector<char> >( max_depth );
-      memset( &bi, char(0), sizeof(bi) );
+      //memset( &bi, char(0), sizeof(bi) );
+      bi = sha512();
```

Comment:
This is inefficient. Please add an ``inline clear() { memset... }`` method to sha512 and use that. Same for the other hash classes.